lib/Dialect/FIRRTL/Transforms/GrandCentralSignalMappings.cpp

```diff
@@ -249,11 +274,13 @@ FModuleOp ModuleSignalMappings::emitMappingsModule() {
     // llvm::errs() << "Rewriting " << circuitName << " " << pathName <<
     // "\nWith: " << markDut << " " << Prefix << "\n";
     bool seenRoot = false;
+    auto [modulePath, varPath] = pathName.split('>');
     if (markDut.empty()) {
-      remoteXmrName += circuitName.drop_front();
+      // If no DUT, top-level is first module in modulePath
+      // ("~Top|Foo/b:B" -> "Foo.b")
+      remoteXmrName += modulePath.split('/').first;
       seenRoot = true;
     }
-    auto [modulePath, varPath] = pathName.split('>');
     do {
       auto [item, tail] = modulePath.split(':');
       modulePath = tail;
```

Comment:
This may able to be dramatically simplified by using `circt::firrtl::tokenizePath` as opposed to making updates to the existing hand-rolled target logic that was already present.

Reply:
Good call, thanks! Replaced this logic with simpler code using `tokenizePath`, thanks! :+1: